web/source/keymanweb.js

```diff
@@ -3058,15 +3076,91 @@
     Lscript.charset="UTF-8";        // KMEW-89
-    Lscript.src = keymanweb.getKeyboardPath(kbdFile);       
     Lscript.type = 'text/javascript';
+
+    var kbdFile = kbdStub['KF'];
+    var kbdLang = kbdStub['KL'];
+    var kbdName = kbdStub['KN'];
+
+    if(typeof(kbdStub.asyncLoader.callback) == "function") {
+      // Add a handler for cases where the new <script> block fails to load.
+      Lscript.addEventListener('error', function() {
+        // Clear the timeout timer.
+        window.clearTimeout(kbdStub.asyncLoader.timer);
+        kbdStub.asyncLoader.timer = null;
+
+        // We already know the load has failed... why wait?
+        kbdStub.asyncLoader.callback('Cannot find the ' + kbdName + ' keyboard for ' + kbdLang + '!');
+        console.log('Error:  cannot find the', kbdName, 'keyboard for', kbdLang, 'at', kbdFile + '!');
+
+        kbdStub.asyncLoader = null;
+      }, true);
+    }
+    
+    // The load event will activate a newly-loaded keyboard if successful and report an error if it is not.
+    Lscript.addEventListener('load', function() {
+      if(typeof(kbdStub.asyncLoader.callback) == "function") {
+        // Clear the timeout timer.
+        window.clearTimeout(kbdStub.asyncLoader.timer);
+        kbdStub.asyncLoader.timer = null;
+      }
+
+      // To determine if the load was successful, we'll need to check the keyboard array for our desired keyboard.
+      //FIXME:
+      // Test if keyboard already loaded
+      var kbd, Li;
+      for(Li=0; Li<keymanweb._Keyboards.length; Li++) {
+        if(kbdStub['KI'] == keymanweb._Keyboards[Li]['KI']) {
+          kbd = keymanweb._Keyboards[Li];
+        } 
+      }
+
+      if(kbd != undefined) {  // Is cleared upon a successful load.
+        
+        //Activate keyboard, if it's still the active stub.
+        if(kbdStub == keymanweb._ActiveStub) {
+          keymanweb.doBeforeKeyboardChange(kbd['KI'],kbdStub['KLC']);
+          keymanweb._ActiveKeyboard=kbd;
+
+          if(keymanweb._LastActiveElement != null) 
+          {
+            keymanweb._JustActivatedKeymanWebUI = 1;
+            keymanweb._SetTargDir(keymanweb._LastActiveElement);            
+          }
+
+          String.kmwEnableSupplementaryPlane(kbdStub && ((kbdStub['KS'] && (kbdStub['KS'] == 1)) || (kbd['KN'] == 'Hieroglyphic'))); // I3319 - SMP extension, I3363 (Build 301)
+          keymanweb.saveCurrentKeyboard(kbd['KI'],kbdStub['KLC']);
+        
+          // Prepare and show the OSK for this keyboard
+          osk._Load();
+
+          // Remove the wait message, if defined
+          if(typeof (util.wait) == 'function') {
+            util.wait(false);
+          }
+        }
+      } else {
+        // A handler portion for cases where the new <script> block loads, but fails to process.
+        if(typeof(kbdStub.asyncLoader.callback) == "function") {
+
+          kbdStub.asyncLoader.callback('Error registering the ' + kbdName + ' keyboard for ' + kbdLang + '!');
```

Comment:
Let's not use exclamation marks in the error messages. I know we did previously, but I don't think we need them. Here and elsewhere :)